server/api/controllers/config/environments/environmentsConfigController.js

```diff
@@ -59,9 +67,49 @@ function putEnvironmentConfigByName(req, res, next) {
  * DELETE /config/environments/{name}
  */
 function deleteEnvironmentConfigByName(req, res, next) {
-  const key = req.swagger.params.name.value;
+  const environmentName = req.swagger.params.name.value;
   const user = req.user;
-  return dynamoHelper.delete(key, user).then(_ => res.status(200).end()).catch(next);
+  
+  return co(function* () {
+    let accountName = yield Environment.getAccountNameForEnvironment(environmentName);
+
+    yield [
+      deleteLBSettingsForEnvironment(environmentName, accountName, user),
+      deleteLBUpstreamsForEnvironment(environmentName, accountName, user)
+    ];
+
+    yield deleteEnvironment(environmentName, accountName, user);
+    res.status(200).end();
+  }).catch(next);
+}
+
+function deleteLBSettingsForEnvironment(environmentName, accountName, user) {
+  return co(function*(){
+    let lbSettingsList = yield lbSettingsTable.queryRangeByKey(environmentName);
+    return lbSettingsList.map(lbSettings => {
+      return lbSettingsTable.deleteWithSortKey(environmentName, lbSettings.VHostName, user, { accountName });
+    });
+  });
+}
+
+function deleteLBUpstreamsForEnvironment(environmentName, accountName, user) {
+  return co(function*(){
+    let allLBUpstreams = yield lbUpstreamsTable.getAll(null, { accountName });
+    let lbUpstreams = allLBUpstreams.filter(lbUpstream => {
+      return lbUpstream.Value.EnvironmentName.toLowerCase() === environmentName.toLowerCase();
```

Comment:
I don't like this, do we have to allow users to case environment however they want? it's environment name, not a Windows file system :-)